newrelic_rpm.gemspec

```diff
@@ -47,6 +47,7 @@ https://github.com/newrelic/newrelic-ruby-agent/
   s.homepage = "https://github.com/newrelic/rpm"
   s.require_paths = ["lib"]
   s.summary = "New Relic Ruby Agent"
+  s.add_development_dependency 'feedjira', '3.2.1' unless ENV['CI'] # for Gabby
```

Comment:
The 'feedjira' and 'httparty' addition are saying that we don't want them installed if we're running the CI? But I would've thought that was the only place we'd need them. Can you help explain?

Reply:
I'm also curious about this! I'll hold off final approval until you have a chance to answer.

Reply:
> The 'feedjira' and 'httparty' addition are saying that we don't want them installed if we're running the CI? But I would've thought that was the only place we'd need them. Can you help explain?

Yeah, good callout here @hannahramadan.

- Developers iterating on Gabby need to run the unit tests located at `test/new_relic/gem_notifications_tests.rb`. These tests require both `httparty` and `feedjira` to work. As such, it's appropriate that they exist as dev dependencies in the `.gemspec` file.
    - `httparty` was already present in `.gemspec` due to some fetching behavior that the perf tests are apparently capable of
    - `feedjira` became needed after Gabby started tracking CVEs
- The installation of `feedjira` causes issues with Ruby 2.2. For now, given that the gem notification tests are not currently exercised by our CI, I've decided to save us on trouble and time by not having the gems get installed.
- Gabby's own work performed as GitHub Actions jobs has no dependency on the Ruby agent, and never ends up installing the agent. As a result, the `.gemspec` file is never brought in, and its declaration of `httparty` and `feedjira` as dev dependencies is ignored. Gabby's GHA jobs use `gem install` commands directly to bring these gems in.
- Long term, I believe that Gabby should be moved out into a separate gem, complete with a separate `.gemspec` file, and a separate CI setup that exercises the unit tests. For now though, these changes give us a tiny boost in performance and simplification by stopping the installation of `httparty` when running CI, and gives devs the necessary `feedjira` gem for local dev.